example/spec/testAPI/ExecutionStatusSpec.js

```diff
@@ -0,0 +1,153 @@
+
+  const updatedInputPayloadJson = globalReplace(inputPayloadJson, testDataGranuleId, granuleId);
+
+  return JSON.parse(updatedInputPayloadJson);
+}
+
+describe('The Cumulus API ExecutionStatus', () => {
+  let workflowExecution = null;
+  const collection = { name: 'MOD09GQ', version: '006' };
+  const provider = { id: 's3_provider' };
+  const inputPayloadFilename = './spec/testAPI/testAPI.input.payload.json';
+  let inputPayload;
+  let granuleId;
+  process.env.ExecutionsTable = `${config.stackName}-ExecutionsTable`;
+  process.env.GranulesTable = `${config.stackName}-GranulesTable`;
+  process.env.UsersTable = `${config.stackName}-UsersTable`;
+
+  beforeAll(async () => {
+    granuleId = randomStringFromRegex(granuleRegex);
+
+    const inputPayloadJson = fs.readFileSync(inputPayloadFilename, 'utf8');
+    inputPayload = await setupTestGranuleForAPI(config.bucket, granuleId, inputPayloadJson);
+
+    workflowExecution = await buildAndExecuteWorkflow(
+      config.stackName, config.bucket, workflowName, collection, provider, inputPayload
+    );
+  });
+
+  afterAll(async () => {
+    // Remove the granule files added for the test
+    await Promise.all(
+      inputPayload.granules[0].files.map((file) =>
+        s3().deleteObject({
+          Bucket: config.bucket, Key: `${file.path}/${file.name}`
+        }).promise())
+    );
+  });
+
+  it('completes execution with success status', () => {
+    expect(workflowExecution.status).toEqual('SUCCEEDED');
+  });
+
+  describe('When accessing a workflow execution via the API', () => {
+    let executionStatus;
+
+    beforeAll(async () => {
+      const executionArn = workflowExecution.executionArn;
+      executionStatus = await apiTestUtils.getExecutionStatus({
+        prefix: config.stackName,
+        arn: executionArn
+      });
+    });
+
+    it('returns the inputs and outputs for the entire workflow', async () => {
+      expect(executionStatus.execution).toBeTruthy();
+      expect(executionStatus.execution.executionArn).toEqual(workflowExecution.executionArn);
+      const input = JSON.parse(executionStatus.execution.input);
+      const output = JSON.parse(executionStatus.execution.output);
+      expect(input.payload).toEqual(inputPayload);
+      expect(output.payload || output.replace).toBeTruthy();
+    });
+
+    it('returns the stateMachine information and workflow definition', async () => {
+      expect(executionStatus.stateMachine).toBeTruthy();
+      expect(executionStatus.stateMachine.stateMachineArn).toEqual(executionStatus.execution.stateMachineArn);
+      expect(executionStatus.stateMachine.stateMachineArn.endsWith(executionStatus.stateMachine.name)).toBe(true);
+
+      const definition = JSON.parse(executionStatus.stateMachine.definition);
+      expect(definition.Comment).toEqual('Ingest Granule');
+      const stateNames = Object.keys(definition.States);
+
+      const expectedStates = ['Report', 'VpcOrNot', 'SyncGranule', 'SyncGranuleNoVpc', 'ChooseProcess',
```

Comment:
If the workflow changes it will break this test. I wonder if there is some way to grab it from the config? At least comment maybe in the workflows.yml that if the states change, this needs to be updated.

Reply:
👍